tests/ci/integration/run_socat_integration.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+# SRC_ROOT(aws-lc)
+#  - SCRATCH_FOLDER
+#    - SOCAT_SRC
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SRC_ROOT}/"scratch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+SOCAT_SRC="${SCRATCH_FOLDER}/socat"
+
+function build_and_test_socat() {
+  pushd "$SOCAT_SRC"
+  autoconf
+  ./configure --enable-openssl-base="$AWS_LC_INSTALL_FOLDER"
+  make -j "$NUM_CPU_THREADS"
+  # test 146 OPENSSLLISTENDSA: fails because AWS-LC doesn't support FFDH ciphersuites which are needed for DSA
+  # test 216 UDP6MULTICAST_UNIDIR: known flaky test in socat with newer kernels
+  # test 309 OPENSSLRENEG1: AWS-LC doesn't support renegotiation by default, it can be enabled by calling SSL_set_renegotiate_mode
+  # but that has caveats. See PORTING.md 'TLS renegotiation'
+  # test 310 OPENSSLRENEG2: AWS-LC doesn't support renegotiation by default, it can be enabled by calling SSL_set_renegotiate_mode
+  # but that has caveats. See PORTING.md 'TLS renegotiation'
+  # test 399 OPENSSL_DTLS_CLIENT: Unknown issue running openssl s_server
+  # test 467 EXEC_FDS: Something broken with exec'ing and not inheriting LD_LIBRARY_PATH
+  # test 468 EXEC_SNIFF: Something broken with exec'ing and not inheriting LD_LIBRARY_PATH
+  # test 478 SIGUSR1_STATISTICS: GHA does not support tty
+  # test 492 ACCEPT_FD: uses systemd-socket-activate which doesn't inherit the LD_LIBRARY_PATH so socat can't find libcrypto.so
+  # test 498 SHELL_SOCKETPAIR: GHA does not specify expected shell environment variables
```

Comment:
would linking LC statically resolve these issues?

Reply:
That's an interesting idea. I like the `ldd` sanity check but we could build twice and run a different subset of tests. Shared only took 6 minutes.

Reply:
Tried it [here](https://github.com/aws/aws-lc/actions/runs/7616553807/job/20743582946?pr=1387) the only difference is `492 ACCEPT_FD` passed which was expected to avoid the LD_LIBRARY_PATH. 467/468 still fail with something in the exec. And all the 500-530 tests still fail due to the shell variables. I'm going to skip the static test since I don't think it's worth the compute for the one extra test.